rclpy/rclpy/node.py

```diff
@@ -160,6 +160,12 @@ def destroy_node(self):
         ret = True
         if self.handle is None:
             return ret
+
+        # ensure that the passed node contains a valid capsule
+        class_ = self.handle.__class__
+        if not class_ or class_.__name__ != 'PyCapsule':
+            raise ValueError()
```

Comment:
It was only used in one bizarre test (which modified the internal handle of a node instance) so I thought it isn't a real world case. Added a message in f2b9bf5ee1b5874fc6540754dca71f41934af68d.